nixos/modules/services/hardware/fancontrol.nix

```
@@ -0,0 +1,46 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.hardware.fancontrol;
+  configFile = pkgs.writeText "fan.conf" cfg.config;
+
+in {
+
+  options.hardware.fancontrol = {
+    enable = mkEnableOption "fancontrol (requires fancontrol.config)";
```

```suggestion
    enable = mkEnableOption "fancontrol";
```
That a config is needed should be now become obvious from the following options.

---

i don't usually read the actual module but use nixos.org/nixos/options.html, it doesn't seem obvious to me in that context, neither is the error when not providing a config

---

Once you look under nixos.org/nixos/options.html for `hardware.fancontrol` you will see two options. To me that would be obvious(?) but it is also OK to leave the hint there.

---

i think `hardware.fancontrol.enable = true;` is so obvious it's easy to not look further
i agree that most people would figure it out, but if i've got fans screaming at me, i'd want all the help i can get